src/graph/optimizer/OptRule.cpp

```diff
@@ -31,6 +31,13 @@ const PlanNode *MatchedResult::planNode(const std::vector<int32_t> &pos) const {
   return DCHECK_NOTNULL(result->node)->node();
 }
 
+void MatchedResult::collectBoundary(std::vector<OptGroup *> &boundary) const {
```

Comment:
It's better to collect all `OptGroupNode` rather than `OptGroup` in the path of matched result.

Reply:
But the `dependencies_` is `OptGroup`.